homeassistant/components/recorder/models/state.py

```diff
@@ -143,14 +144,14 @@ def row_to_compressed_state(
     entity_id: str,
     state: str,
     last_updated_ts: float | None,
+    no_attributes: bool,
 ) -> dict[str, Any]:
     """Convert a database row to a compressed state schema 41 and later."""
-    comp_state: dict[str, Any] = {
-        COMPRESSED_STATE_STATE: state,
-        COMPRESSED_STATE_ATTRIBUTES: decode_attributes_from_source(
+    comp_state: dict[str, Any] = {COMPRESSED_STATE_STATE: state}
+    if not no_attributes:
+        comp_state[COMPRESSED_STATE_ATTRIBUTES] = decode_attributes_from_source(
             getattr(row, "attributes", None), attr_cache
-        ),
-    }
```

Comment:
This was triggering a column now found exception which was now obvious when it was re-sorted.

It all worked but we shouldn't have sqlalchemy raising and than capturing it